dysnix/heimdall/templates/statefulset.yaml

```diff
@@ -0,0 +1,184 @@
+      {{- end }}
+      - name: {{ template "heimdall.name" . }}
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+        {{- with .Values.securityContext }}
+        securityContext:
+          {{- toYaml . | nindent 10 }}
+        {{- end }}
+        {{- with .Values.command }}
+        command:
+          {{- toYaml . | nindent 10 }}
+        {{- end }} 
+        args:
+          - start
+          - --home
+          - /data
+          - --rest-server
+          {{- with .Values.extraArgs }}
+          {{- toYaml . | nindent 10 }}
+          {{- end }}
+        ports:
+        {{- range $k, $v := .Values.containerPorts }}
+        {{- if $v.enabled }}
+          - name: {{ $k }}
+            containerPort: {{ $v.port }}
+            {{- with $v.protocol }}
+            protocol: {{ . }}
+            {{- end }}
+            {{- with $v.hostPort }}
+            hostPort: {{ . }}
+            {{- end }}
+        {{- end }}
+        {{- end }}
+        {{- if .Values.livenessProbe.enabled }}
+        livenessProbe:
+          {{- if and .Values.livenessProbe.exec (kindIs "string" .Values.livenessProbe.exec.command) }}
+          {{- omit .Values.livenessProbe "enabled" "exec" | toYaml | nindent 10 }}
+          exec:
+            command:
+              {{- tpl .Values.livenessProbe.exec.command . | nindent 14 }}
+          {{- else }}
+          {{- omit .Values.livenessProbe "enabled" | toYaml | nindent 10 }}
+          {{- end }}
+        {{- end }}
+        {{- if .Values.readinessProbe.enabled }}
+        readinessProbe:
+          {{- if and .Values.readinessProbe.exec (kindIs "string" .Values.readinessProbe.exec.command) }}
+          {{- omit .Values.readinessProbe "enabled" "exec" | toYaml | nindent 10 }}
+          exec:
+            command:
+              {{- tpl .Values.readinessProbe.exec.command . | nindent 14 }}
+          {{- else }}
+          {{- omit .Values.readinessProbe "enabled" | toYaml | nindent 10 }}
+          {{- end }}
+        {{- end }}
+        volumeMounts:
+        - name: data
+          mountPath: /data
+        - name: config
+          mountPath: /data/config/config.toml
+          subPath: config.toml
+        - name: config
+          mountPath: /data/config/heimdall-config.toml
+          subPath: heimdall-config.toml
+        - name: config
+          mountPath: /data/config/app.toml
+          subPath: app.toml
+        resources:
+          {{- toYaml .Values.resources | nindent 10 }}
+      {{- with .Values.sidecarContainers }}
+      {{- toYaml . | nindent 6 }}
+      {{- end }}
+      volumes:
+        - name: config
+          configMap:
+            name: {{ include "heimdall.fullname" . }}-config
+        - name: scripts
+          configMap:
+            name: {{ include "heimdall.fullname" . }}-scripts
+        {{- if eq .Values.persistence.type "hostPath" }}
```

Comment:
Did we miss `pvc` handling here alongside hostpath ?

Reply:
It's my mistake, I  expected that we need to specify `volume` explicitly inside `volumes` even it's created via `volumeClaimTemplates`. But it's auto-populated.